sound/soc/sof/intel/mtl.c

```diff
@@ -255,9 +256,13 @@ static int mtl_dsp_pre_fw_run(struct snd_sof_dev *sdev)
 	if (ret < 0)
 		dev_err(sdev->dev, "failed to power up gated DSP domain\n");
 
-	/* make sure SoundWire is not power-gated */
-	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, MTL_HFPWRCTL,
-				MTL_HfPWRCTL_WPIOXPG(1), MTL_HfPWRCTL_WPIOXPG(1));
+	hdev = sdev->pdata->hw_pdata;
+	/* Check scanned SoundWire capabilities exposed by DSDT */
+	if (hdev->info.handle && hdev->info.link_mask > 0)
+		/* make sure SoundWire is not power-gated */
```

Comment:
This one comment with an update might be enough?
`/* if SoundWire is used, make sure it is not power-gated */`

Reply:
Ack